probe-rs/src/debug/debug_info.rs

```diff
@@ -103,13 +103,14 @@ impl DebugInfo {
     /// This function will currently return the innermost function in that case.
     pub fn function_name(
         &self,
+        memory: &mut impl MemoryInterface,
```

Comment:
It was required to get rid of an option down the line which made all the code complex.

Reply:
But now the interface is weird, there shouldn't be any need to have memory access to use this function. And it makes testing harder.

Reply:
I find it weirder if one passes an Option that then can be None somewhere 6 layers deep and weird shit happens because of it and all code is more complicated.

Reply:
I added a comment. I think it can be factored out by assuming that memory never has to be read for this case. I would like to do this when we have more tests in place as I don't know if that assumption is correct.

Also, this function name is misleading, because it gets the function name of a function at a specific memory location which, I think, it sometimes has to resolve. We have a different function with the same name that gets the function name from a die.